### DISCUSSION TITLE
Defining types that can be reused in models

### DISCUSSION THREAD

**Discussion by thdxr** [prisma/prisma#4846] [Q&A]
Was hoping to have a discussion around the scenario of commonly used types that can be shared between multiple models.  For example
```
model Customer {
  address_street String
  address_city   String
  address_state  String
}

model BusinessOwner {
  home_address_street String
  home_address_city   String
  home_address_state  String
  
  business_address_street String
  business_address_city   String
  business_address_state  String
}
```

I've forked the `prisma-engines` repo to experiment with some ways to make this more concise and implemented the following

```
partial Address {
  street String
  city   String
  state  String
}

model Customer {
  address Address
}

model BusinessOwner {
  home Address
  business Address
}
```

I implemented this purely as syntactic sugar on the AST and it's really just a pre-processor.  I was curious if the Prisma team has thought about providing a solution for this scenario or if the current thinking is that this is out of scope or it's better to be explicit and copy paste. This also _could_ be mixed in with the concept of `embed` which I saw some code for

**Comment by thdxr** [comment]
Really this is just expanding the undocumented type alias feature to allow specifying multiple types and perhaps a custom prefix

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
